File: 102-binary-tree-level-order-traversal/102-binary-tree-level-order-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        
        
        vector<vector<int>> ans;
        
        if(root == nullptr)
            return ans; 
        
        
        queue<TreeNode*> q; 
        
        q.push(root); 
        
        
        while(q.empty() == false)
        {
  
            int count = q.size(); 
            
            vector<int> res; 
            for(int i = 0; i < count; i++)
            {
                
                TreeNode* curr = q.front(); 
                q.pop(); 
                
                
                res.push_back(curr->val); 
                
                if(curr->left != nullptr)
                    q.push(curr->left); 
                
                if(curr->right != nullptr)
                    q.push(curr->right); 
                
        
            }
            
            ans.push_back(res); 
    
        }
            
        return ans;     
        
    }
};

File: 102-binary-tree-level-order-traversal/102-binary-tree-level-order-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        Queue<TreeNode> q = new LinkedList<>();
        List<List<Integer>> ans = new ArrayList<>(); 
        if(root == null)
            return ans;
        q.add(root);
        while(q.size() !=0){
            int count = q.size();
            List<Integer> res = new ArrayList<>(); 
            for(int i=0;i<count; i++){
                TreeNode curr = q.remove();//first element will be deleted
                //  System.out.println(curr.val);
                res.add(curr.val); 
                 if(curr.left != null){
                    q.add(curr.left);
                 }
                 if(curr.right !=null){
                     q.add(curr.right);
                 }
            }
            ans.add(res);
        }

        return ans;
    }
}